### ISSUE TITLE
Rule table-groups may return false positives

### ISSUE THREAD

**Issue by pichfl** [ember-template-lint/ember-template-lint#464]
This yields an error:

```hbs
<table>
  ...
  {{#if foo}}
    <tfoot>
      <tr>...</tr>
    </tfoot>
  {{/if}}
</table>
```

This does not

```hbs
<table>
  ...
  <tfoot>
    {{#if foo}}
      <tr>...</tr>
    {{/if}}
  </tfoot>
</table>
```

While I'm not sure the rule can make exceptions like this, I'd still prefer dropping the `<tfoot>` if empty instead of leaving it in the dom.

**Comment by pichfl** [comment]
I'll try to get something in by Monday 🙂

**Comment by rwjblue** [comment]
OK, after thinking about this some, I think we should specifically allow `if` / `unless` and begin validating _their_ children (as if they were in the outer level directly).

**Comment by rwjblue** [comment]
Fixed by https://github.com/ember-template-lint/ember-template-lint/pull/638
